anyway/views/news_flash/api.py

```diff
@@ -281,3 +282,34 @@ def normalize_query_param(key, value: list):
 def normalize_query(params: dict):
     params_non_flat = params.to_dict(flat=False)
     return {k: normalize_query_param(k, v) for k, v in params_non_flat.items()}
+
+
+@roles_accepted(BE_CONST.Roles2Names.Authenticated.value)
+def update_new_flash_qualifying(id):
```

Comment:
I think I forgot to mention some details :(
- Only `manual` qualification can come with location. The code should check that, and issue an error if location arrives with a different qualification, or `manual` qualification arrives without a location.
In general, if the invocation does not succeeds, it should issue an error message (both to the log and to the caller).

Reply:
done